### PR TITLE
feat: add backfill.yaml template to baseline_clients_daily_v1 and logic to render it

### PR DESCRIPTION
**Pull request by kik-kik**
# feat: add backfill.yaml template to baseline_clients_daily_v1 and logic to render it

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-3846)

### CODE REVIEW

File: sql_generators/glean_usage/templates/baseline_clients_daily_v1.backfill.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{{ header_yaml }}
+{% if app_name == "fenix" -%}
```
Comment (kik-kik) [inline sql_generators/glean_usage/templates/baseline_clients_daily_v1.backfill.yaml:2]: This is to make sure if we ever add more backfill configuration we do not accidentally add this entry to other app ids.

**Comment by ANich** [comment]
Confirming that these are the tables that you want to be backfilled @kik-kik? from the [integration report](https://github.com/mozilla/bigquery-etl/pull/5626#issuecomment-2122671625)

```
Only in /tmp/workspace/generated-sql/sql/moz-fx-data-shared-prod/org_mozilla_fenix_derived/baseline_clients_daily_v1: backfill.yaml
Only in /tmp/workspace/generated-sql/sql/moz-fx-data-shared-prod/org_mozilla_fenix_nightly_derived/baseline_clients_daily_v1: backfill.yaml
Only in /tmp/workspace/generated-sql/sql/moz-fx-data-shared-prod/org_mozilla_fennec_aurora_derived/baseline_clients_daily_v1: backfill.yaml
Only in /tmp/workspace/generated-sql/sql/moz-fx-data-shared-prod/org_mozilla_firefox_beta_derived/baseline_clients_daily_v1: backfill.yaml
Only in /tmp/workspace/generated-sql/sql/moz-fx-data-shared-prod/org_mozilla_firefox_derived/baseline_clients_daily_v1: backfill.yaml
```

File: sql_generators/glean_usage/common.py [hunk 2]
```diff
@@ -313,6 +326,15 @@ def generate_per_app_id(
             if schema:
                 artifacts.append(Artifact(table, "schema.yaml", schema))
 
+            if backfill_config:
+                if app_name not in APP_WITH_BACKFILL_CONFIG:
+                    logging.info(
+                        "Skipped copying backfill.yaml for %s as app: %s is not expected to contain any backfill configuration."
+                        % (table, app_name)
```
[the remark this answers is not in the document]
Comment (kik-kik) [reply sql_generators/glean_usage/common.py:333]: unless something has changed, % is lazily evaluated and therefore preferred option for logging?

File: sql_generators/glean_usage/common.py
[the remark this answers is not in the document]
Comment (kik-kik) [reply sql_generators/glean_usage/common.py:1]: That could be an option too, we'd just have to manually add all 5 of the directories to `sql/moz-fx...`.

**Comment by kik-kik** [comment]
Closing in favour of: https://github.com/mozilla/bigquery-etl/pull/5628

**Pull request closed without merging**